Source/Csla/Configuration/Fluent/CslaDataPortalConfiguration.cs

```diff
@@ -31,13 +45,33 @@ public static CslaDataPortalConfiguration DataPortal(this ICslaConfiguration con
     /// <param name="builder"></param>
     public static CslaDataPortalConfiguration AddServerSideDataPortal(this CslaDataPortalConfiguration builder)
     {
+      return AddServerSideDataPortal(builder, null);
+    }
+
+    /// <summary>
+    /// Add services required to host the server-side
+    /// data portal.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="options"></param>
+    public static CslaDataPortalConfiguration AddServerSideDataPortal(this CslaDataPortalConfiguration builder, Action<DataPortalServerOptions> options)
```

Comment:
This method was getting called automatically from the CslaConfigurationExtensions type's AddCsla method. I haven't seen this removed, so I wonder if we may get unusual behaviour from it being called twice - and with different instances of the configuration data, as the DataPortal() extension method returns a new instance of its type on each call.

This behaviour of the DataPortal() method caused me some pain when I was working on my solution, as it led to some nasty bugs in my code.

Reply:
You are right, in that in this latest bit of code I put in some Add methods instead of TryAdd. I've changed those now, so the second call (if there is one) of the `AddServerSideDataPortal` method doesn't really do anything.